### PR TITLE
refactor(react-grid): replace render functions with components in TableView

### PR DESCRIPTION
**Pull request by kvet**
BREAKING CHANGES:

The TableView's `tableLayoutTemplate`, `tableCellTemplate`, `tableRowTemplate`, `tableNoDataCellTemplate`, `tableNoDataRowTemplate`, `tableStubCellTemplate`, and `tableStubHeaderCellComponent` properties have been replaced with `layoutComponent`, `getCellComponent`, `rowComponent`, `noDataCellComponent`, `noDataRowComponent`, `stubCellComponent` and `stubHeaderCellComponent`. This also means that they accept components instead of render functions. Find more details here: https://github.com/DevExpress/devextreme-reactive/issues/496

### CODE REVIEW

File: packages/dx-react-grid-bootstrap3/src/templates/table-layout.test.jsx [hunk 1]
```diff
@@ -21,25 +21,25 @@ describe('TableLayout', () => {
     jest.resetAllMocks();
   });
 
-  it('should pass rowTemplate to TableLayoutCore', () => {
+  it('should pass rowComponent to TableLayoutCore', () => {
     const rowTemplateMock = jest.fn();
     const rowTemplateArgs = {
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-layout.test.jsx]: Inconsistent names. It seems like they should contain *Component.

File: packages/dx-react-grid/docs/reference/table.md [hunk 2]
```diff
@@ -58,69 +59,69 @@ rowId? | number &#124; string | Specifies the associated row's ID.
...
+Describes properties of a table column the Table plugin renders.
 
 A value with the following shape:
 
 Field | Type | Description
 ------|------|------------
 key | string | A table column's unique identifier.
 type | string | Specifies the table column type. The specified value defines which cell template is used to render the column.
-column? | [Column](#column) | Specifies the associated user column.
+column? | [Column](#column-extension) | Specifies the associated user column.
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/docs/reference/table.md]: It looks like you removed all '-' symbols from link anchors, but not here.

File: packages/dx-react-grid-bootstrap3/src/templates/virtual-table-layout.jsx [hunk 3]
```diff
@@ -9,18 +9,16 @@ import { Table } from './table';
...
 const MINIMAL_COLUMN_WIDTH = 120;
 
-const containerTemplate = props => <TableContainer {...props} />;
-const headTableTemplate = props => <Table use="head" {...props} />;
-const tableTemplate = props => <Table {...props} />;
-const headTemplate = props => <thead {...props} />;
-const bodyTemplate = props => <tbody {...props} />;
+const HeadTable = props => <Table use="head" {...props} />;
+const TableHead = props => <thead {...props} />;
+const TableBody = props => <thead {...props} />;
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid-bootstrap3/src/templates/virtual-table-layout.jsx]: `<thead {...props} />` -> `<tbody {...props} />` ?

File: packages/dx-react-grid/docs/reference/table.md (hunk 2 above)
Comment (kvet) [reply packages/dx-react-grid/docs/reference/table.md]: It's a way markdown generate default header ids.

**Pull request merged**